synapse/handlers/federation.py

```diff
@@ -276,10 +278,6 @@ async def _maybe_backfill_inner(
             insertion_events_to_be_backfilled,
         )
 
-        if not backwards_extremities and not insertion_events_to_be_backfilled:
-            logger.debug("Not backfilling as no extremeties found.")
-            return False
-
         # we now have a list of potential places to backpaginate from. We prefer to
         # start with the most recent (ie, max depth), so let's sort the list.
         sorted_backfill_points: List[_BackfillPoint] = sorted(
```

Comment:
Can we do anything smarter with two already sorted lists?

Reply:
There's [`heapq.merge`](https://docs.python.org/3/library/heapq.html#heapq.merge) in the stdlib which runs in linear time. Though it probably doesn't make much difference for any reasonable number of extremities.